zkevm-circuits/src/evm_circuit.rs

```diff
@@ -197,11 +197,12 @@ impl<F: Field> EvmCircuit<F> {
         }
     }
 
-    pub fn new_dev(block: Block<F>, fixed_table_tags: Vec<FixedTableTag>) -> Self {
-        Self {
-            block: Some(block),
-            fixed_table_tags,
-        }
+    // TODO move this impl into test-mod, until have a better way to
+    // get_required_num_rows of EvmCircuit
+    pub fn get_num_rows_required(block: &Block<F>) -> usize {
+        let mut cs = ConstraintSystem::default();
+        let config = EvmCircuit::<F>::configure(&mut cs);
+        config.get_num_rows_required(block)
```

Comment:
This https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1039 blocks gathering EvmCircuit test-related into test-mod.

Reply:
Not sure I follow. What's the issue with this??

Reply:
> This #1039 blocks gathering EvmCircuit test-related into test-mod.

Apology for confusing you. I'll detail it.

For now, when we invoke `get_num_rows_required` of evm-circuit, we have to invoke  the `EvmCircuit::<F>::configure`. However, the `EvmCircuit::<F>::configure` is in `impl Circuit for EvmCircuit`. 
As #953 desc, we'd better hide the `impl Circuit for EvmCircuit` by moving it into test mod. 
So it's little block this pr.

Reply:
Ohh I see! Let's wait there and feel free to continue with this once it's merged!!!

Reply:
But where do we need to call `get_num_rows_required` that is not in the tests? I'm not sure if this is really blocking something.

Reply:
Oh.yeah. I got it(parm-face). 

I‘ve resolved the conflicts. Please review.